tests/v2/validation/rbac/rbac_psa_test.go

```diff
@@ -0,0 +1,336 @@
+		errStatus := strings.Split(err.Error(), ".")[1]
+		rgx := regexp.MustCompile(`\[(.*?)\]`)
+		errorMsg := rgx.FindStringSubmatch(errStatus)
+		assert.Equal(rb.T(), "403 Forbidden", errorMsg[1])
+
+	}
+
+	rb.T().Logf("Validate creation of new namespace with PSA labels as %v", role)
+
+	labels = map[string]string{
+		psaWarn:    pssBaselinePolicy,
+		psaEnforce: pssBaselinePolicy,
+		psaAudit:   pssBaselinePolicy,
+	}
+	namespaceName := namegenerator.AppendRandomString("testns-")
+	namespaceCreate, err := namespaces.CreateNamespace(rb.nonAdminUserClient, namespaceName, "{}", labels, map[string]string{}, rb.adminProject)
+
+	switch role {
+	case restrictedAdmin, roleOwner:
+		require.NoError(rb.T(), err)
+		expectedLabels := getPSALabels(response, labels)
+		assert.Equal(rb.T(), labels, expectedLabels)
+		_, err = createDeployAndWait(rb.steveNonAdminClient, rb.client, rb.cluster.ID, "nginx", "nginx", namespaceCreate.Name)
+		require.NoError(rb.T(), err)
+	case roleMember, roleProjectOwner, roleProjectMember, roleProjectReadOnly:
+		require.Error(rb.T(), err)
+		if role == roleMember || role == roleProjectReadOnly {
+			errMessage := strings.Split(err.Error(), ":")[0]
+			assert.Equal(rb.T(), "Resource type [namespace] is not creatable", errMessage)
+		} else {
+			errStatus := strings.Split(err.Error(), ".")[1]
+			rgx := regexp.MustCompile(`\[(.*?)\]`)
+			errorMsg := rgx.FindStringSubmatch(errStatus)
+			assert.Equal(rb.T(), "403 Forbidden", errorMsg[1])
+		}
+	}
+
+}
+
+func (rb *PSATestSuite) ValidateAdditionalPSA(role string) {
+	createProjectAsNonAdmin, err := createProject(rb.nonAdminUserClient, rb.cluster.ID)
+	require.NoError(rb.T(), err)
+
+	namespaceName := namegen.AppendRandomString("testns-")
+	createNamespace, err := namespaces.CreateNamespace(rb.nonAdminUserClient, namespaceName, "{}",
+		map[string]string{}, map[string]string{},
+		createProjectAsNonAdmin)
+	require.NoError(rb.T(), err)
+
+	rb.T().Logf("Validate editing new namespace in a cluster member created project with PSA labels as %v", role)
+	labels := map[string]string{
+		psaWarn:    pssRestrictedPolicy,
+		psaEnforce: pssRestrictedPolicy,
+		psaAudit:   pssRestrictedPolicy,
+	}
+
+	updateNS := rb.ConvertNS(createNamespace)
+	updateNS.Labels = labels
+
+	relogin, err := rb.nonAdminUserClient.ReLogin()
+	require.NoError(rb.T(), err)
+	rb.nonAdminUserClient = relogin
+
+	steveStdUserclient, err := rb.nonAdminUserClient.Steve.ProxyDownstream(rb.cluster.ID)
+	require.NoError(rb.T(), err)
+	rb.steveNonAdminClient = steveStdUserclient
+
+	response, err := rb.steveNonAdminClient.SteveType(namespaces.NamespaceSteveType).Update(createNamespace, updateNS)
+
+	switch role {
+	case roleOwner:
+		require.NoError(rb.T(), err)
+		expectedLabels := getPSALabels(response, labels)
+		assert.Equal(rb.T(), labels, expectedLabels)
+		_, err = createDeployAndWait(rb.steveNonAdminClient, rb.client, rb.cluster.ID, "nginx", "nginx", createNamespace.Name)
```

Comment:
Nit from the previous PR, I think `nginx` strings could be constant as well, or if the same `psaNginx` name is used, could be updated to the constant